### PR TITLE
ci: implement GitHub Actions CI

### PR DESCRIPTION
**Pull request by icyphox**
## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

## Further comments

This PR implements CI in GitHub Actions, transitioning away from CircleCI. There are related changes in [ory/ci](https://github.com/ory/ci) as well. Below are the required secrets for the workflow to function:

- `MAILCHIMP_API_KEY`
- `SLACK_WEBHOOK_URL` — configure this in [Incoming Webhooks](https://api.slack.com/incoming-webhooks).
- `GORELEASER_KEY` — key for GoReleaser Pro.

This also requires the creation of an [environment](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) called `production`. This environment needs to be configured with [required reviewers](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#required-reviewers). This ensures that the `newsletter-send` job is protected behind manual approval. Also see: https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments

NOTE: This PR depends on https://github.com/ory/ci/pull/59 for the `releaser` and `newsletter` actions.

### CODE REVIEW

**Comment by aeneasr** [comment]
When you test this, it will send out the newsletter if I set up `MAILCHIMP_API_KEY`. We need to ensure that it uses a test mailing list with e.g. you and me on it.

**Comment by icyphox** [comment]
@aeneasr Correct. Can you setup a temp list and share its list & segment IDs with me?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    name: Draft newsletter
+    runs-on: ubuntu-latest
+    if: ${{ github.ref_type == 'tag' }}
+    needs:
+      - release
+    steps:
+      - uses: ory/ci/newsletter@v1
+        with:
+          mailchimp_list_id: 058a056618
+          mailchmip_segment_id: 11398953
```
Comment (aeneasr) [inline .github/workflows/ci.yaml:74]: I confirmed that these values are indeed the a test audience - good to go @icyphox

**Comment by aeneasr** [comment]
All set up

**Comment by icyphox** [comment]
Closing this and reopening a new PR as I had erroneously made this from a fork, which I deleted.

**Pull request closed without merging**